core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -79,16 +79,16 @@ public void onExpiration() {
      */
     @Override
     public void onComplete() {
-        log.trace("Completing the delayed share fetch request for group {}, member {}, " +
-                        "topic partitions {}", shareFetchData.groupId(),
-            shareFetchData.memberId(), shareFetchData.partitionMaxBytes().keySet());
+        log.trace("Completing the delayed share fetch request for group {}, member {}, "
+            + "topic partitions {}", shareFetchData.groupId(), shareFetchData.memberId(),
+            topicPartitionDataFromTryComplete != null ? topicPartitionDataFromTryComplete.keySet() : "null");
 
         if (shareFetchData.future().isDone())
             return;
 
         Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData;
         // tryComplete did not invoke forceComplete, so we need to check if we have any partitions to fetch.
-        if (topicPartitionDataFromTryComplete.isEmpty())
+        if (topicPartitionDataFromTryComplete == null || topicPartitionDataFromTryComplete.isEmpty())
```

Comment:
For my understanding, how can this value be null? We initialize it in the `DelayedShareFetch` constructor and its updation always returns a map.

Reply:
I read one thing wrong and instead of null it'a always empty. Re-checking and verifying why this fixed consistently with this PR change, I can easily reproduce the issue without this change and can never with. I will update.

Reply:
I have added tests.